### PR TITLE
feat: support CA bundle and identity cert/key authentication

### PR DESCRIPTION
**Pull request by Teng-Jiao-Chen**
### Description of your changes
Provide an solution for community to support CA auth

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- Kind Cluster e2e workflow on README is verified working, no existing workflow get affected.
- Integ test is performed in LinkedIn environment
- Helm yaml generation is verified

```
The default behavior has no changes.

For the yamls with CA info enabled
==================================================

helm template member-agent ./charts/member-agent/ \
--set useCAAuth=true \
--set config.identityKey="identity-key-path" \
--set config.identityCert="identity-cert-path" \
--set config.CABundle="ca-bundle-path" \
--set config.hubURL=hub-cluster-api-url \
--set image.repository=member-agent-image-url \
--set image.tag=0.0.1 \
--set refreshtoken.repository=refresh-token-url \
--set refreshtoken.tag=0.0.1 \
--set image.pullPolicy=IfNotPresent --set refreshtoken.pullPolicy=IfNotPresent \
--set config.memberClusterName="member-cluster-name" \
--set logVerbosity=5 \
--set namespace=fleet-system \
| awk -vout=/tmp/member-agent-with-ca -F": " '$0~/^# Source: /{file=out"/"$2; print "Creating "file; system ("mkdir -p $(dirname "file"); echo "" > "file)} $0!~/^#/ && $0!="---"{print $0 >> file}'

deployment.yaml is verified with the right arguments and there is no refresh token generated
==================================================
...
          args:
            - --leader-elect=true
            - --use-ca-auth=true
            - --v=5
            - -add_dir_header
          env:
...
          - name: IDENTITY_KEY
            value:  "identity-key-path"
          - name: IDENTITY_CERT
            value:  "identity-cert-path"
          - name: CA_BUNDLE
            value:  "ca-bundle-path"
...
```

### CODE REVIEW

**Comment by Teng-Jiao-Chen** [comment]
@microsoft-github-policy-service agree [company="LinkedIn"]

**Comment by Teng-Jiao-Chen** [comment]
@microsoft-github-policy-service agree company="LinkedIn"

File: cmd/memberagent/main.go [hunk 1]
```diff
@@ -72,7 +73,7 @@ func main() {
 	}
 	tokenFilePath := os.Getenv("CONFIG_PATH")
 
-	if tokenFilePath == "" {
+	if !*useCaAuth && tokenFilePath == "" {
 		klog.ErrorS(errors.New("hub token file path cannot be empty"), "error has occurred retrieving CONFIG_PATH")
```
Comment (Ealianis) [inline cmd/memberagent/main.go:77]: Error message no longer reflects that neither the token file path nor useCaAuth bundle is set to be used.

File: cmd/memberagent/main.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ import (
 
 var (
 	scheme               = runtime.NewScheme()
+	useCaAuth            = flag.Bool("use-ca-auth", false, "Use identity and CA bundle to authenticate the member agent.")
```
Comment (Ealianis) [inline cmd/memberagent/main.go:42]: nit: Stylistically Go lang convention for acronyms is to match the casing used, e.g., `useCAAuth`.

Comment (Teng-Jiao-Chen) [reply cmd/memberagent/main.go]: Thanks, changed ;-)

File: cmd/memberagent/main.go (hunk 1 above)
Comment (Teng-Jiao-Chen) [reply cmd/memberagent/main.go]: Thanks for catching it up. Update the error msg to reflect the CA auth part.

**Review by michaelawyu — DISMISSED** [review]
LGTM.

File: cmd/memberagent/main.go [hunk 3]
```diff
@@ -85,21 +86,51 @@ func main() {
...
-		// an error if there is no file.
-		_, err := os.Stat(tokenFilePath)
-		return err
-	})
-	if err != nil {
-		klog.ErrorS(err, " cannot retrieve token file from the path %s", tokenFilePath)
-		os.Exit(1)
+	identityKeyFile := os.Getenv("IDENTITY_KEY")
+	identityCertFile := os.Getenv("IDENTITY_CERT")
+	cABundleFile := os.Getenv("CA_BUNDLE")
```
Comment (zhiying-lin) [inline cmd/memberagent/main.go:91]:
```suggestion
	caBundleFile := os.Getenv("CA_BUNDLE")
```
Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case. 
https://github.com/golang/go/wiki/CodeReviewComments#initialisms

**Review by zhiying-lin — COMMENTED** [review]
one minor comment, others LGTM

Can you change the PR title to "feat: support CA bundle and identity cert/key authentication"? i remember we have the pr title lint check and not sure why the current one did not fail.

Thank you so much!

**Comment by Ealianis** [comment]
> one minor comment, others LGTM
> 
> Can you change the PR title to "feat: support CA bundle and identity cert/key authentication"? i remember we have the pr title lint check and not sure why the current one did not fail.
> 
> Thank you so much!

That is odd, I triggered it manually.

File: cmd/memberagent/main.go (hunk 3 above)
Comment (Teng-Jiao-Chen) [reply cmd/memberagent/main.go]: Change the title and `ca*` suggested 

Thanks for the review ;-)

File: charts/member-agent/templates/deployment.yaml [hunk 4]
```diff
@@ -63,15 +75,16 @@ spec:
           volumeMounts:
           - name: provider-token 
             mountPath: /config
```
Comment (zhiying-lin) [inline charts/member-agent/templates/deployment.yaml:77]: do we need this volumn when CAAuth is enabled?

Comment (Teng-Jiao-Chen) [reply charts/member-agent/templates/deployment.yaml:78]: Thanks for catching this up.

- changed the scope of the conditional bracket to ignore `provider-token` mounts and tested on local

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**